File: movieApp/app/src/main/java/com/lbc/practice/movieapp/view/second/SecondActivity.java
package com.lbc.practice.movieapp.view.second;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.lbc.practice.movieapp.model.PlayingMovieResult;
import com.lbc.practice.movieapp.R;
import com.lbc.practice.movieapp.adapter.ImagePagerAdapter;

/**
 * Created by lbc on 2018-04-02.
 */

public class SecondActivity extends AppCompatActivity implements ViewPager.OnPageChangeListener,SecondContract.SecondView{

    private ViewPager viewPager;
    private ImagePagerAdapter imagePagerAdapter;
    private TextView tvPagerCounter, tvTitle, tvGrade, tvStory;
    private int page;
    SecondPresenter secondPresenter;

    PlayingMovieResult.ResultsBean resultsBean;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        tvPagerCounter = (TextView)findViewById(R.id.second_tv_viewpager_counter);
        tvTitle = (TextView)findViewById(R.id.second_tv_title);
        tvGrade = (TextView)findViewById(R.id.second_tv_grade);
        tvStory = (TextView)findViewById(R.id.second_tv_story);

        secondPresenter = new SecondPresenter(this);
        Intent intent =getIntent();
        resultsBean = (PlayingMovieResult.ResultsBean) intent.getSerializableExtra("movieInfo");

        String img[] = new String[2];
        img[0] = "http://image.tmdb.org/t/p/w185"+resultsBean.getPoster_path();
        img[1] = "http://image.tmdb.org/t/p/w185"+resultsBean.getBackdrop_path();

        viewPager = (ViewPager)findViewById(R.id.second_viewpager);
        imagePagerAdapter = new ImagePagerAdapter();
        viewPager.setAdapter(imagePagerAdapter);
        viewPager.addOnPageChangeListener(this);
        imagePagerAdapter.setImageUrls(img);
        secondPresenter.displayViewPage(1);
        imagePagerAdapter.notifyDataSetChanged();

        tvTitle.setText(resultsBean.getTitle());
        if (resultsBean.isAdult()) {
            tvGrade.setText("성인 등급");
        } else {
            tvGrade.setText("일반 등급");
        }

        tvStory.setText(resultsBean.getOverview());

    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
        if(positionOffset > 0.5){
            if(page != position + 2)
                secondPresenter.displayViewPage(position+2);
        }else if(page != position + 1){
            secondPresenter.displayViewPage(position+1);
        }
    }

    @Override
    public void onPageSelected(int position) {

    }

    @Override
    public void onPageScrollStateChanged(int state) {
        if(state == ViewPager.SCROLL_STATE_IDLE){
            secondPresenter.displayViewPage(viewPager.getCurrentItem()+1);
        }
    }

    @Override
    public void setPage(int page) {
        this.page = page;
    }

    @Override
    public void setText(int page) {
        tvPagerCounter.setText(page+"/"+2);
    }
}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/view/second/SecondActivity.kt
package com.lbc.practice.movieapp.view.second;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.lbc.practice.movieapp.R;
import com.lbc.practice.movieapp.adapter.ImagePagerAdapter;
import com.lbc.practice.movieapp.data.PlayingMovieResult;
import kotlinx.android.synthetic.main.activity_second.*

/**
 * Created by lbc on 2018-04-02.
 */

class SecondActivity : AppCompatActivity(), ViewPager.OnPageChangeListener, SecondContract.SecondView {

    private var imagePagerAdapter: ImagePagerAdapter? = null
    private var tvPagerCounter: TextView? = null

    private var page: Int = 0
     var secondPresenter: SecondPresenter?=null

     var resultsBean: PlayingMovieResult.ResultsBean?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        secondPresenter = SecondPresenter(this)
        val intent = intent
        resultsBean = intent.getSerializableExtra("movieInfo") as PlayingMovieResult.ResultsBean


        val img = arrayOfNulls<String>(2)
        img[0] = "http://image.tmdb.org/t/p/w185" + resultsBean?.poster_path
        img[1] = "http://image.tmdb.org/t/p/w185" + resultsBean?.backdrop_path

        imagePagerAdapter = ImagePagerAdapter()
        second_viewpager!!.adapter = imagePagerAdapter
        second_viewpager!!.addOnPageChangeListener(this)
//        imagePagerAdapter!!.setImageUrls(img)
        imagePagerAdapter!!.imageUrls =img
        secondPresenter?.displayViewPage(1)
        imagePagerAdapter!!.notifyDataSetChanged()

        second_tv_title!!.text = resultsBean?.title
        if (resultsBean!!.isAdult) {
            second_tv_grade!!.text = "성인 등급"
        } else {
            second_tv_grade!!.text = "일반 등급"
        }

        second_tv_story!!.text = resultsBean?.overview

    }

    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
        if (positionOffset > 0.5) {
            if (page != position + 2)
                secondPresenter?.displayViewPage(position + 2)
        } else if (page != position + 1) {
            secondPresenter?.displayViewPage(position + 1)
        }
    }

    override fun onPageSelected(position: Int) {

    }

    override fun onPageScrollStateChanged(state: Int) {
        if (state == ViewPager.SCROLL_STATE_IDLE) {
            secondPresenter?.displayViewPage(second_viewpager!!.currentItem + 1)
        }
    }

    override fun setPage(page: Int) {
        this.page = page
    }

    override fun setText(page: Int) {
        second_tv_viewpager_counter!!.text = page.toString() + "/" + 2
    }
}
